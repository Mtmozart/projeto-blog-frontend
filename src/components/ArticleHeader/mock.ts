import { ArticleHeaderProps } from '.';
import { data } from '../../api/data.json';

const { id } = data.posts.data[0];
const { title, excerpt, cover, author, category } =
  data.posts.data[0].attributes;

export default {
  id,
  title,
  excerpt,
  cover,
  author,
  category,
  createdAt: '2023-06-02T15:16:09.246+00:00',
} as ArticleHeaderProps;
