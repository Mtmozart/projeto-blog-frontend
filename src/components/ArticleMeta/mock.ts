import { ArticleMetaProps } from '.';
import { data } from '../../api/data.json';
const { author, categories } = data.posts.data[0].attributes;
export default {
  createdAt: '2023-06-02T15:16:09.246+00:00',
  author,
  categories,
} as ArticleMetaProps;
