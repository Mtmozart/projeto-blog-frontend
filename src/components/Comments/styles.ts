import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: 0 ${theme.spacings.large};
  `}
`;
