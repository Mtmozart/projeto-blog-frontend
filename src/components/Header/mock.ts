import { HeaderProps } from '.';

export default {
  blogName: 'Fallen Calvo',
  blogDescription:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud.',
  logo: {
    data: {
      identificador: '3',
      attributes: {
        name: '3334910-desktop-source-code-and-wallpaper-by-coding-and-programming-gratis-foto.jpg',
        altText: null,
        url: 'https://pbs.twimg.com/media/FqKWLbAWwAAM7KA?format=jpg&name=small',
      },
    },
  },
  showText: true,
} as HeaderProps;
