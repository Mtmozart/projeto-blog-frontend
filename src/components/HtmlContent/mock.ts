export default `
  <p style="text-align:justify;">&nbsp;</p>
  <p style="text-align:justify;">O React 6 chegou com uma série de recursos e melhorias que prometem levar o desenvolvimento de interfaces de usuário a um novo patamar. Neste artigo, exploraremos as principais novidades do React 6 e discutiremos a importância dessa atualização para a comunidade de desenvolvedores.</p>
  <p style="text-align:justify;"><strong>1. Hooks reimaginados: Aumentando a flexibilidade</strong> Os Hooks foram introduzidos no React 5 e revolucionaram a forma como lidamos com estado e efeitos colaterais. No React 6, os Hooks foram reimaginados, permitindo uma maior flexibilidade na composição de lógica e comportamento em componentes funcionais. Agora, podemos criar Hooks personalizados mais poderosos e reutilizáveis,
  simplificando ainda mais o desenvolvimento de aplicações.</p>
  <p style="text-align:justify;"><strong>2. Performance aprimorada: Mais rápido do que nunca</strong>
  A equipe do React focou em melhorias de desempenho no React 6. A nova versão apresenta um algoritmo de
  reconciliação aprimorado e otimizações inteligentes de renderização. Isso resulta em uma experiência mais
  rápida para os usuários finais, com atualizações de interface suaves e tempos de carregamento reduzidos.</p>
  <p style="text-align:justify;"><strong>3. Novas APIs de contexto: Compartilhamento de estado facilitado</strong>
  O React 6 introduz APIs de contexto aprimoradas, facilitando o compartilhamento de estado entre componentes.
   Agora, podemos criar contextos mais flexíveis e granulares,
  permitindo um gerenciamento de estado mais eficiente e evitando problemas comuns de prop drilling
  (passagem excessiva de propriedades).</p><p style="text-align:justify;"><strong>4. Suporte nativo a TypeScript:
  Desenvolvimento robusto e seguro</strong> O TypeScript tem ganhado popularidade na comunidade de desenvolvimento
   JavaScript, e o React 6 abraçou essa tendência. Agora, o React oferece suporte nativo a TypeScript, permitindo
    o desenvolvimento de aplicativos mais robustos e seguros. A integração perfeita entre o React e o TypeScript
    oferece autocompletar, detecção de erros em tempo de compilação e uma melhor experiência de desenvolvimento e
    m geral.</p><p style="text-align:justify;"><strong>Conclusão: O futuro do desenvolvimento de interfaces</strong>
     Com o React 6, a biblioteca JavaScript continua a evoluir e redefinir a forma como construímos interfaces de
     usuário. Os novos recursos e melhorias trazem mais flexibilidade, desempenho aprimorado, compartilhamento de
     estado facilitado e suporte nativo a TypeScript. Essas mudanças tornam o desenvolvimento de aplicações mais
     eficiente, seguro e agradável. O React 6 marca um marco importante na história do React e confirma sua posição
     como uma das principais escolhas para o desenvolvimento de interfaces modernas e reativas. Não perca tempo e
     comece a explorar o React 6 hoje mesmo!</p><figure class="image"><img alt="alien-planet.jpg"
     src="http://localhost:1337/uploads/alien_planet_a3727af9d6.jpg" srcset="http://localhost:1337/uploads/alien_planet_a3727af9d6.jpg 245w, /uploads/small_alien_planet_a3727af9d6.jpg 500w, /uploads/medium_alien_planet_a3727af9d6.jpg 750w, /uploads/large_alien_planet_a3727af9d6.jpg 1000w" sizes="100vw" width="1000">
     </figure><p style="text-align:justify;">&nbsp;</p><p style="text-align:justify;">&nbsp;</p>
`;
