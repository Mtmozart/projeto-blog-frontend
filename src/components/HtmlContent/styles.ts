import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: calc(${theme.font.sizes.small} + 0.2rem);
    line-height: 1.5;
    p {
      margin: ${theme.spacings.medium} 0;
    }

    a,
    a:visited,
    a:link {
      color: ${theme.colorsThemes.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }
    code {
      font-family: monospace;
      color: ${theme.colorsThemes.secondary};
      font-size: ${theme.font.sizes.small};
      background: ${theme.colorsThemes.mediumGray};
      padding: 0.2rem;
      margin: 0.2rem;
    }

    pre {
      background: ${theme.colorsThemes.primary};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colorsThemes.white};
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
      margin: ${theme.spacings.medium} 0;
    }

    pre code {
      color: inherit;
      background: inherit;
    }
    img {
      max-width: 100%;
    }

    .image {
      ${theme.colorsThemes.mediumGray}
      line-height: 0;
      margin: ${theme.spacings.medium} 0;
    }
    .image figcaption {
      font-size: ${theme.font.sizes.small}
      padding: ${theme.spacings.small}
      text-align: center;
      ine-height: 1.3;
    }

    .image-style-side {
      float: right;
      max-width: 50%;
      margin: ${theme.spacings.medium} 0;
    }
    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colorsThemes.mediumGray};
    }

    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }
    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }
    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
    }
    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colorsThemes.mediumGray};
    }
    blockquote {
      border-left: 0.5rem solid ${theme.colorsThemes.secondary};
      color: ${theme.colorsThemes.darkerGray};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }

      @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        float: none;
        max-width: 100%;
        margin: ${theme.spacings.medium}
      }
    }

  `}
`;
