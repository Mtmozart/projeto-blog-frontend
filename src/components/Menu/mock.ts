import { MenuProps } from '.';

export default {
  links: [
    {
      id: '1',
      link: '#home',
      text: 'Home',
      newTab: false,
    },
    {
      id: '2',
      link: '#contact',
      text: 'contato',
      newTab: false,
    },
    {
      id: '3',
      link: '#posts',
      text: 'Posts',
      newTab: false,
    },
    {
      id: '4',
      link: 'https://hotmart.com/',
      text: 'Cursos',
      newTab: true,
    },
    {
      id: '5',
      link: '#home',
      text: 'Home',
      newTab: false,
    },
    {
      id: '6',
      link: '#contact',
      text: 'contato',
      newTab: false,
    },
    {
      id: '7',
      link: '#posts',
      text: 'Posts',
      newTab: false,
    },
    {
      id: '8',
      link: 'https://hotmart.com/',
      text: 'Cursos',
      newTab: true,
    },
  ],
  blogName: 'Matheus Mozart',
  logo: 'https://lumiere-a.akamaihd.net/v1/images/alderaan-main_f5b676cf.jpeg?region=0%2C0%2C1280%2C720',
} as MenuProps;
