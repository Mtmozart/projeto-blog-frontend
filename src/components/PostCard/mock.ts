import { data } from '../../api/data.json';

const { id } = data.posts.data[0];
const {
  title,
  slug,
  content,
  excerpt,
  cover,
  allowComments,
  author,
  categories,
} = data.posts.data[0].attributes;
const post = {
  id,
  title,
  slug,
  content,
  excerpt,
  cover,
  allowComments,
  author,
  categories,
};

export default post;
