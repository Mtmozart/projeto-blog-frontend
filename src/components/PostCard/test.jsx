import { screen } from '@testing-library/react';
import { PostCard } from '.';
import { renderTheme } from '../../styles/render-theme';
import mock from './mock';

const props = mock;

describe('<PostCard />', () => {
  it('should render a heading, cover and excerpt', () => {
    renderTheme(<PostCard {...props} />);
  });
  expect(
    screen.getByRole('heading', { name: props.title }),
  ).toBeInTheDocument();
});
