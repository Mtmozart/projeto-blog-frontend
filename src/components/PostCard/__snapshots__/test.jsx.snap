// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
.c3 {
  color: #000000;
  font-size: 2.4rem;
  text-transform: none;
}

.c0 .c2 {
  margin: 0;
  margin-top: calc(1.6rem -0.6rem);
  margin-bottom: 1.6rem;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover a {
  color: #dc143c;
}

.c0:hover img {
  opacity: 0.8;
}

.c1 {
  max-width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/post/a-importancia-do-react-6-no-frontend"
    >
      <img
        alt="A importância do react 6 no frontend"
        class="c1"
        src="https://img.freepik.com/vetores-gratis/cidade-moderna-no-rio-a-noite_1441-2829.jpg?w=1060&t=st=1687883386~exp=1687883986~hmac=8f2d3eb84e58e071e05d2506ae05451b54aa8f23e648beaf0f684ecc8d1f795f"
      />
    </a>
    <h2
      class="c2 c3"
    >
      <a
        href="/post/a-importancia-do-react-6-no-frontend"
      >
        A importância do react 6 no frontend
      </a>
    </h2>
    <p
      class=""
    >
      Bem-vindo ao nosso site sobre React! Aqui você encontrará um resumo conciso e abrangente sobre essa popular biblioteca JavaScript para construção de interfaces de usuário.
    </p>
  </div>
</div>
`;
